File: common/net/cazzar/mods/jukeboxreloaded/gui/GuiHandler.java
package net.cazzar.mods.jukeboxreloaded.gui;

import cpw.mods.fml.common.network.IGuiHandler;
import net.cazzar.mods.jukeboxreloaded.blocks.TileJukebox;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;

public class GuiHandler implements IGuiHandler {
    public static final int JUKEBOX = 0;

    @Override
    public Object getClientGuiElement(int ID, EntityPlayer player, World world,
                                      int x, int y, int z) {
        final TileEntity tile = world.getBlockTileEntity(x, y, z);

        switch (ID) {
            case JUKEBOX:
                if (!(tile instanceof TileJukebox)) return null;
                return new GUIJukebox(player, (TileJukebox) tile);
            default:
                return null;
        }
    }

    @Override
    public Object getServerGuiElement(int ID, EntityPlayer player, World world,
                                      int x, int y, int z) {
        final TileEntity tile = world.getBlockTileEntity(x, y, z);

        switch (ID) {
            case JUKEBOX:
                if (!(tile instanceof TileJukebox)) return null;
                return new ContainerJukebox(player.inventory,
                        (TileJukebox) tile);
            default:
                return null;
        }
    }

}

File: src/main/scala/net/cazzar/mods/jukeboxreloaded/common/gui/GuiHandler.scala
package net.cazzar.mods.jukeboxreloaded.common.gui

import cpw.mods.fml.common.network.IGuiHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World
import net.cazzar.mods.jukeboxreloaded.client.GuiJukebox
import net.cazzar.mods.jukeboxreloaded.blocks.tileentity.TileJukebox

object GuiHandler extends IGuiHandler{
    val JUKEBOX: Int = 0

    def getServerGuiElement(id: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = id match {
        case JUKEBOX => {
            val tile: TileJukebox = world.func_147438_o(x, y, z).asInstanceOf[TileJukebox]
            new ContainerJukebox(player.inventory, tile)
        }
    }

    def getClientGuiElement(id: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = id match {
        case JUKEBOX => {
            val tile: TileJukebox = world.func_147438_o(x, y, z).asInstanceOf[TileJukebox]
            new GuiJukebox(player, tile)
        }
    }
}
